src/antidote.erl

```diff
@@ -36,22 +55,134 @@
+
+-spec read_objects(Objects::[bound_object()], TxId::txid())
+                  -> {ok, [term()]} | {error, reason()}.
+read_objects(Objects, TxId) ->
+    %%TODO: Transaction co-ordinator handles multiple reads
+    %% Executes each read as in a interactive transaction
+    Results = lists:map(fun({Key, Type, _Bucket}) ->
+                                case clocksi_iread(TxId, Key, Type) of
+                                    {ok, Res} ->
+                                        Res;
+                                    {error, _Reason} ->
+                                        error
+                                end
+                        end, Objects),
+    case lists:member(error, Results) of
+        true -> {error, read_failed}; %% TODO: Capture the reason for error
+        false -> {ok, Results}
+    end.
+
+-spec update_objects([{bound_object(), op(), op_param()}], txid())
+                    -> ok | {error, reason()}.
+update_objects(Updates, TxId) ->
+    %% TODO: How to generate Actor,
+    %% Actor ID must be removed from crdt update interface
+    Actor = TxId,
+    %% Execute each update as in an interactive transaction
+    Results = lists:map(
+                fun({{Key, Type, _Bucket}, Op, OpParam}) ->
+                        case clocksi_iupdate(TxId, Key, Type,
+                                             {{Op, OpParam}, Actor}) of
+                            ok -> ok;
+                            {error, _Reason} ->
+                                error
+                        end
+                end, Updates),
+    case lists:member(error, Results) of
+        true -> {error, read_failed}; %% TODO: Capture the reason for error
+        false -> ok
+    end.
+
+%% For static transactions: bulk updates and bulk reads
+-spec update_objects(snapshot_time(), term(), [{bound_object(), op(), op_param()}]) ->
+                            {ok, snapshot_time()} | {error, reason()}.
+update_objects(Clock, _Properties, Updates) ->
+    Actor = actor, %% TODO: generate unique actors
+    Operations = lists:map(
+                   fun({{Key, Type, _Bucket}, Op, OpParam}) ->
+                           {update, {Key, Type, {{Op,OpParam}, Actor}}}
+                   end,
+                   Updates),
+    case clocksi_execute_tx(Clock, Operations) of
+        {ok, {_TxId, [], CommitTime}} ->
+            {ok, CommitTime};
+        {error, Reason} -> {error, Reason}
+    end.
+
+read_objects(Clock, _Properties, Objects) ->
+    Args = lists:map(
+             fun({Key, Type, _Bucket}) ->
+                     {read, {Key, Type}}
+             end,
+             Objects),
+    case clocksi_execute_tx(Clock, Args) of
+        {ok, {_TxId, Result, CommitTime}} ->
+            {ok, Result, CommitTime};
+        {error, Reason} -> {error, Reason}
+    end.
+
+
+%% Object creation and types
+-spec create_bucket(bucket(), type()) -> {ok, bucket()}.
+create_bucket(Bucket, _Type) ->
+    %% TODO: Bucket is not currently supported
```

Comment:
Shouldn't this yield a currently_not_supported error as with the other not-yet implemented features?